src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -14,14 +14,16 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed class BindingDiagnosticBag : BindingDiagnosticBag<AssemblySymbol>
     {
+        private static readonly ObjectPool<BindingDiagnosticBag> s_poolWithBoth = new ObjectPool<BindingDiagnosticBag>(() => new BindingDiagnosticBag(s_poolWithBoth!, new DiagnosticBag(), new HashSet<AssemblySymbol>()));
+        private static readonly ObjectPool<BindingDiagnosticBag> s_poolWithDiagnosticsOnly = new ObjectPool<BindingDiagnosticBag>(() => new BindingDiagnosticBag(s_poolWithDiagnosticsOnly!, new DiagnosticBag(), dependenciesBag: null));
+        private static readonly ObjectPool<BindingDiagnosticBag> s_poolWithDependenciesOnly = new ObjectPool<BindingDiagnosticBag>(() => new BindingDiagnosticBag(s_poolWithDependenciesOnly!, diagnosticBag: null, new HashSet<AssemblySymbol>()));
+
         public static readonly BindingDiagnosticBag Discarded = new BindingDiagnosticBag(null, null);
 
-        public BindingDiagnosticBag()
-            : this(usePool: false)
-        { }
+        private readonly ObjectPool<BindingDiagnosticBag>? _pool;
 
-        private BindingDiagnosticBag(bool usePool)
-            : base(usePool)
+        public BindingDiagnosticBag()
```

Comment:
There aren't many usages of the public constructor. Do you know why those were not converted those to `GetInstance`+`Free`?